operations/exponentOperation.h

```diff
@@ -1,6 +1,6 @@
 /*
-This file defines the ExponentOperation class which represents the
-exponentiation of a tensor.
+    This file defines the ExponentOperation class which represents the
+    exponentiation of a tensor.
```

Comment:
General comment about this operation - We will end up allocating pointer space for one more tensor pointer here (t2) because it extends from "Operation" which declares 3 pointers in memory.

One possible solution to avoiding this can be to have two different types of operations extending from the base "Operation" class - "BinaryOperation" and "UnaryOperation" and then having our operations extending from either of these. I can take that up as a separate issue if you want.

Reply:
Yeah, we could do this but I think it'll add unnecessary complexity to the code. I think we should keep it the way it is. I'm all for memory saving but also want to keep the code to as few lines as possible. One tensor pointer takes really minimal space so I think we should be good.

Reply:
Alright, that's fair.